CONTRIBUTING.md

```diff
@@ -1,5 +1,46 @@
+### Setup
+
+We recommend to create a virtual environment to contribute to the client. You can install all developer dependencies using `pip install requirements.txt`.
+
+### Testing
+
+We use [pytest](https://docs.pytest.org) to write tests, however there are plenty of older tests that use unittest.
+
+There are three kinds of tests:
+- Unit tests, that test individual components of the client
+- Integration tests, that test the client with a running weaviate instance
+- Mock tests, where a weaviate instance is mocked to return specific replies
+
+To run the integration tests start the weaviate instances using `./ci/start_weaviate.sh`.
+Then run all tests with
+```
+pytest integration
```

Comment:
I just did this:
```
./ci/start_weaviate.sh
pytest integration
```

However I'm getting a lot of failures, see partial output:
```
integration/test_authentication.py:40: WeaviateStartUpError                                                            __________ test_authentication_user_pw[okta - no refresh-test@test.de-OKTA_DUMMY_CI_PW-8083-some_scope-True] __________                                                                                                                       
recwarn = WarningsRecorder(record=True), name = 'okta - no refresh', user = 'test@test.de'                             
env_variable_name = 'OKTA_DUMMY_CI_PW', port = '8083', scope = 'some_scope', warning = True      
```

So looks like some step is missing?

Reply:
Could you double check in docker if all containers are running?

Reply:
@databyjp confirmed that the steps are working for a new setup, so I'll merge this. Just mesasge me on slack if you have further problems :)